aviatrix/resource_aviatrix_remote_syslog.go

```diff
@@ -0,0 +1,179 @@
+				Optional:    true,
+				Description: "Server Public Certificate: Public certificate of the controller signed by the same CA",
+			},
+			"private_key_file_path": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Server Private Key: Private key of the controller that pairs with the public certificate",
+			},
+			"protocol": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "TCP",
+				ValidateFunc: validation.StringInSlice([]string{"TCP", "UDP"}, false),
+				Description:  "Protocol: TCP or UDP (TCP by default)",
+			},
+			"template": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Optional Custom Template: Useful when forwarding to 3rd party servers like Datadog or Sumo",
+			},
+			"excluded_gateways": {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Description: "List of excluded gateways.",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"notls": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "True if not protected by TLS.",
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Enabled or not.",
+			},
+		},
+	}
+}
+
+func resourceAviatrixRemoteSyslogCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*goaviatrix.Client)
+
+	remoteSyslog := &goaviatrix.RemoteSyslog{
+		Server:            d.Get("server").(string),
+		Port:              d.Get("port").(int),
+		Protocol:          d.Get("protocol").(string),
+		Index:             d.Get("index").(int),
+		Template:          d.Get("template").(string),
+		CaCertificate:     d.Get("ca_certificate_file_path").(string),
+		PublicCertificate: d.Get("public_certificate_file_path").(string),
+		PrivateKey:        d.Get("private_key_file_path").(string),
+	}
+
+	if !((remoteSyslog.CaCertificate != "" && remoteSyslog.PublicCertificate != "" && remoteSyslog.PrivateKey != "") ||
+		(remoteSyslog.CaCertificate == "" && remoteSyslog.PublicCertificate == "" && remoteSyslog.PrivateKey == "")) {
+		return fmt.Errorf("one or more certificates missing")
+	}
+
+	var excludeGateways []string
+	for _, v := range d.Get("excluded_gateways").(*schema.Set).List() {
+		excludeGateways = append(excludeGateways, v.(string))
+	}
+	if len(excludeGateways) != 0 {
+		remoteSyslog.ExcludeGatewayInput = strings.Join(excludeGateways, ",")
+	}
+
+	if err := client.EnableRemoteSyslog(remoteSyslog); err != nil {
+		return fmt.Errorf("could not enable remote syslog: %v", err)
+	}
+
+	d.SetId("remote_syslog_" + strconv.Itoa(remoteSyslog.Index))
+	return nil
+}
+
+func resourceAviatrixRemoteSyslogRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*goaviatrix.Client)
+
+	server := d.Get("server").(string)
+
+	if server == "" {
+		id := d.Id()
+		log.Printf("[DEBUG] Looks like an import. Import Id is %s", id)
+		index, _ := strconv.Atoi(id[len(id)-1:])
```

Comment:
You should check this error, the `d.Id()` is a user provided value, there is no guarantee the user used the correct ID format from your documentation. We should return an error like: 
```
fmt.Errorf("invalid ID format, expected ID in format 'remote_syslog_{index} instead got %s", id)
```

Also should check the length of the ID before trying to extract the last character. If ID was an empty string, this would crash.